bhttp.go

```diff
@@ -568,7 +568,7 @@ func UnmarshalBinaryResponse(data []byte) (*http.Response, error) {
 	// Trailer
 	trailerFields := new(fieldList)
 	encodedFieldData, err = readVarintSlice(b)
-	if err != nil {
+	if err != nil && err != io.EOF {
```

Comment:
This is better, but is it sufficient? if `err == io.EOF`, then presumably `len(encodedFieldData) == 0`, and I would expect the following trailer decoding steps to fail. Should we skip them if the trailers are omitted, sort of how the request trailer parsing is skipped?

Also, should we just build a sample client for DOOH to exercise this? Do you have one already?

Reply:
Yes, that looks sufficient. The following decoding steps don't fail, and the trailers map stays empty.

I'm adding DoOH support to `dnscrypt-proxy`, so we'll have a client soon.

Reply:
Awesome! 🎉 

Happy to review that PR when you have it. I assume you'll be rolling your own OHTTP implementation, right? If so, can you please add me the PR? I'd like to make sure we track the implementation alongside others.